rfc/rfc-0007-resource-references.md

```diff
@@ -0,0 +1,40 @@
+# RFC-0007: Resource References
+
+**Authors:** Scott Andrews
+
+**Status:**
+
+**Pull Request URL:** [#1370](https://github.com/projectriff/riff/pull/1370)
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** [RFC 0006: Guidance for child resource names](https://github.com/projectriff/riff/pull/1369)
+
+
+## Problem
+It's common for one resource to reference another. For example, a Deployer can reference a Function build to obtain the latest image, or a stream provider references the gateway Service it created. riff should have a common way to express relationships between Kubernetes resources.
+
+### Anti-Goals
+This RFC is not define how specific riff resources reference other resources, it is only providing guidance.
```

Comment:
While providing guidance is good, I'd suggest being more explicit here about what in riff today is motivating this, and what the likely impact of following these guidelines will be. E.g. It's difficult to infer whether this RFC is talking about issues with container-ref/application-ref/function-ref, or something else.

Reply:
Some RFCs are in the moment, while others are timeless. I've been actively avoiding being too specific about how the RFC applies directly to the current riff codebase, because those references will go stale while the principals will live on. I'm happy to talk about the current state of riff in the comments on the PR.

> It's difficult to infer whether this RFC is talking about issues with container-ref/application-ref/function-ref, or something else.

The current build field on the DeployerSpec/AdapterSpec/ProcessorSpec would be an excellent candidate for being replaced by an ObjectReference. Fields like `*Name` in the status would likewise be a strong candidate for a LocalObjectReference. 

Of course, since this RFC is only providing guidance and not mandating changes, we can approach each of these decisions individually, and more importantly, normalize future behavior.